### PR TITLE
🎓 Add citation cff build target

### PR DESCRIPTION
**Pull request by fwkoch**
[`CITATION.cff`](https://github.com/citation-file-format/citation-file-format/tree/main) is a format [supported by github](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-citation-files) to provide citation metadata, primarily for software/dataset repositories (but also for other `types` including articles!). If a repository contains this file, github will generate APA and bibtex entries.

As MyST projects often include a computational component and often live in github, this citation format is relevant to enable others to cite works in MyST. Since there is significant overlap between `myst.yml` metadata and CFF metadata, it is nice and easy to autogenerate the latter as a MyST export.

### CODE REVIEW

File: packages/myst-cli/src/build/cff.ts [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+  const toc = tic();
+  const { output, articles } = exportOptions;
+  const { clean, projectPath, extraLinkTransformers } = opts ?? {};
+  const article = articles[0];
+  const state = session.store.getState();
+  let frontmatter: PageFrontmatter | undefined;
+  let abstract: string | undefined;
+  if (projectPath && selectors.selectLocalConfigFile(state, projectPath) === sourceFile) {
+    frontmatter = selectors.selectLocalProjectConfig(state, projectPath);
+    const rawAbstract = frontmatter?.parts?.abstract?.join('\n\n');
```
Comment (fwkoch) [inline packages/myst-cli/src/build/cff.ts:290]: This project frontmatter abstract relies on #1396 - until then, this is just `undefined`

File: packages/myst-cli/src/build/cff.ts [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+  vfile.path = output;
+  if (path.basename(output) !== 'CITATION.cff') {
+    fileWarn(
+      vfile,
+      `Invalid Citation File Format filename ${path.basename(output)} - CFF requires filename 'CITATION.cff'`,
+    );
+  }
+  const cff = {
+    ...frontmatterToCFF(frontmatter, abstract),
+    ...exportOptionsToCFF(exportOptions),
```
Comment (fwkoch) [inline packages/myst-cli/src/build/cff.ts:317]: This line allows authors to add (or override) additional CFF info to the export definition, e.g.
```yaml
exports:
  - format: cff
    message: If you cite this article, thank you!!
```

File: packages/myst-cli/src/build/utils/collectExportOptions.ts [hunk 3]
```diff
@@ -312,11 +313,16 @@ export function resolveOutput(
...
   }
   if (!path.extname(output)) {
-    const basename = getDefaultExportFilename(
-      session,
-      singleArticleWithFile(exp.articles)?.file ?? sourceFile,
-      projectPath,
-    );
+    let basename: string;
+    if (exp.format === ExportFormats.cff) {
+      basename = 'CITATION';
```
Comment (fwkoch) [inline packages/myst-cli/src/build/utils/collectExportOptions.ts:318]: The specific filename `CITATION.cff` is part of the spec, so this format does not follow the normal export naming pattern.

File: packages/myst-cli/src/build/cff.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+export async function runCffExport(
+  session: ISession,
+  sourceFile: string,
+  exportOptions: ExportWithOutput,
+  opts?: ExportFnOptions,
+) {
+  const toc = tic();
+  const { output, articles } = exportOptions;
```
Comment (fwkoch) [inline packages/myst-cli/src/build/cff.ts:35]: Default for this `output` should not be in the `_build` folder.

**Review by rowanc1 — APPROVED** [review]

**Review by rowanc1 — APPROVED** [review]
Thanks for adding docs!

**Pull request merged**